### PR TITLE
fix: invalid date_time

### LINKED ISSUE #2261: Invalid DateTime
### Describe the bug
A date field is sometimes displayed as Invalid DateTime.

See Loom: https://www.loom.com/share/bfab89cd438648a2a8f174dc28c5cb0d?sid=2e5b3208-0019-4bc7-86b2-4f5e0ff46310

### System configuration
**Avo version**:
3.1.7

**Rails version**:
7.1.2

**Ruby version**:
3.2.2

**License type**:

 - [ ] Community
 - [ ] Pro
 - [x] Advanced

**Are you using Avo monkey patches, overriding views or view components?**

 - [ ] Yes. If so, please post code samples.
 - [x] No

#### Impact

 - [x] High impact (It makes my app un-usable.)
 - [ ] Medium impact (I'm annoyed, but I'll live.)
 - [ ] Low impact (It's really a tiny thing that I could live with.)

#### Urgency

 - [x] High urgency (I can't continue development without it.)
 - [ ] Medium urgency (I found a workaround, but I'd love to have it fixed.)
 - [ ] Low urgency (It can wait. I just wanted you to know about it.)

### PR DESCRIPTION
**Pull request by Paul-Bob**
# Description

Fields of type `has_one` was eager loading on tabs.

This was causing the `date_time` field's related JS to broke when the field panel was loading on a hidden tab, resulting on the string `Invalid DateTime`.

More specifically the `this.initialValue` on the date field stimulus controller was getting some weird spaces around the value when the field was rendering hidden, we also trim that value now. 

Fixes #2261

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [x] I have added tests that prove my fix is effective or that my feature works

## Manual review steps

1. Step 1
1. Step 2

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

File: lib/avo/fields/concerns/frame_loading.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+        def turbo_frame_loading = kwargs[:turbo_frame_loading]
+
+        def loading
+          turbo_frame_loading || params[:turbo_frame_loading] || "eager"
+        end
+      end
+    end
+  end
+end
```
Comment (Paul-Bob) [inline lib/avo/fields/concerns/frame_loading.rb:16]:
```suggestion
end

```

File: lib/avo/fields/concerns/frame_loading.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+
```
Comment (Paul-Bob) [inline lib/avo/fields/concerns/frame_loading.rb:19]:
```suggestion
end

```

**Review by adrianthedev — CHANGES_REQUESTED** [review]
Could you please add some tests and a scenario where one can try this?

**Comment by Paul-Bob** [comment]
> Could you please add some tests and a scenario where one can try this?

Done, the bug was happening when a `has_one` field was rendering on a secondary tab. Added the comment on user resource that contains a date time field to simulate the bug.

**Comment by Paul-Bob** [comment]
The initial value on JS has some white spaces around it when the field was rendering hidden. We trim that value now

File: app/components/avo/fields/has_one_field/show_component.html.erb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <% if @field.value %>
-  <turbo-frame id="<%= @field.turbo_frame %>" src="<%= @field.frame_url %>" target="_top" class="block">
+  <turbo-frame id="<%= @field.turbo_frame %>" src="<%= @field.frame_url %>"  loading=<%= loading %> target="_top" class="block">
```
Comment (adrianthedev) [inline app/components/avo/fields/has_one_field/show_component.html.erb:2]:
```suggestion
  <turbo-frame id="<%= @field.turbo_frame %>" src="<%= @field.frame_url %>"  loading=<%= turbo_frame_loading %> target="_top" class="block">
```

File: lib/avo/fields/concerns/frame_loading.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  module Fields
+    module Concerns
+      module FrameLoading
+        extend ActiveSupport::Concern
+
+        def turbo_frame_loading = kwargs[:turbo_frame_loading]
+
+        def loading
+          turbo_frame_loading || params[:turbo_frame_loading] || "eager"
+        end
```
Comment (adrianthedev) [inline lib/avo/fields/concerns/frame_loading.rb:11]:
```suggestion
        def turbo_frame_loading
          kwargs[:turbo_frame_loading] || params[:turbo_frame_loading] || "eager"
        end
```

File: lib/avo/fields/concerns/frame_loading.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+module Avo
+  module Fields
+    module Concerns
+      module FrameLoading
+        extend ActiveSupport::Concern
+
+        def turbo_frame_loading
```
Comment (adrianthedev) [inline lib/avo/fields/concerns/frame_loading.rb:7]: Do you think this change will impact any other components @Paul-Bob?

**Review by adrianthedev — APPROVED** [review]

File: lib/avo/fields/concerns/frame_loading.rb (hunk 5 above)
Comment (Paul-Bob) [reply lib/avo/fields/concerns/frame_loading.rb:7]: This code was extracted from `Avo::Fields::HasManyField::ShowComponent` to `Avo::Fields::Concerns::FrameLoading`. 

`Avo::Fields::HasManyField::ShowComponent` includes now `Avo::Fields::Concerns::FrameLoading` so any component that inherit from it and the component itself still have access to the code.

The extraction was made just to reuse the same code inside `Avo::Fields::HasOneField::ShowComponent`.

Any other unrelated components are not being impacted.

Comment (Paul-Bob) [reply lib/avo/fields/concerns/frame_loading.rb:7]: @adrianthedev I misunderstood your question....

Yes, we should also apply this change on `app/components/avo/fields/has_many_field/show_component.html.erb`

**Pull request merged**